pkg/ddc/alluxio/shutdown.go

```diff
@@ -274,6 +279,14 @@ func (e *AlluxioEngine) destroyWorkers(expectedWorkers int32) (currentWorkers in
 			labelNames = append(labelNames, labelExclusiveName)
 		}
 
+		mutex.Lock()
+		currentDataset, err := strconv.Atoi(toUpdate.Labels[labelDatasetNum])
+		if err != nil {
+			return currentWorkers, err
+		}
+		toUpdate.Labels[labelDatasetNum] = strconv.Itoa(currentDataset - 1)
+		mutex.Unlock()
+
 		if len(toUpdate.Labels) < len(node.Labels) {
 			err := e.Client.Update(context.TODO(), toUpdate)
```

Comment:
I think we should `RetryOnConflict` here if the update failed because of resource version conflicts